embrace-android-sdk/embrace-proguard.cfg

```diff
@@ -1,20 +1,33 @@
--keepattributes Exceptions, InnerClasses, Signature, LineNumberTable, SourceFile
+-keepattributes LineNumberTable, SourceFile
 
-## Proguard configuration for Embrace
--keep class io.embrace.android.embracesdk.** { *; }
--dontwarn io.embrace.android.embracesdk.**
+## Keep classes used by hosted SDKs
+-keep class io.embrace.android.embracesdk.Embrace { *; }
+-keep class io.embrace.android.embracesdk.Embrace$AppFramework { *; }
+-keep class io.embrace.android.embracesdk.Embrace$LastRunEndState { *; }
+-keep class io.embrace.android.embracesdk.Severity { *; }
+-keep public class * implements io.embrace.android.embracesdk.internal.EmbraceInternalInterface { *; }
 
-## Proguard configuration for OkHTTP3 / Okio
--dontwarn okhttp3.**
--dontwarn okio.**
-
-## Proguard configuration for Arrow
--keep class java9.** { *; }
--dontwarn java9.**
+## Keep classes used from native code
+-keep class io.embrace.android.embracesdk.payload.NativeThreadAnrSample { *; }
+-keep class io.embrace.android.embracesdk.payload.NativeThreadAnrStackframe { *; }
+-keep class io.embrace.android.embracesdk.anr.sigquit.SigquitDataSource { *; }
```

Comment:
Do we also need to keep `NativeThreadSamplerNdkDelegate`, or will that work ok with R8's obfuscation? It's probably worth global-searching for any `external fun` in the project & checking they work.

Reply:
I verified that they work, but I'm not sure if it's because they're referenced by kept classes or if it's because obfuscation doesn't affect JNI calls. I'll check just in case because I haven't tested every external fun.

The classes referenced from the native code must be kept because they're searched for by name.

Reply:
The classes are always being kept.

Not because we are actually adding any rules for them, but **I think** it's because they are being initialized in modules that override `kotlin.jvm.functions.Function0.invoke()`, and those are also called in different places in ComponentActivity, and every activity of any app is added  because of some aapt_rules.

It's super weird, and I don't want to lose much time on this, so I'll just add rules for the classes that contain `external fun` methods just in case 😅